src/module/actor/mixins/actor-conditions.js

```diff
@@ -1,41 +1,88 @@
+        if (!this._isStatusEffect(conditionName)) {
+            ui.notifications.warn(`Trying to get condition ${conditionName} on actor ${this.name} but the condition is not valid. See CONFIG.SFRPG.statusEffects for all valid conditions.`);
             return undefined;
         }
 
-        const conditionItem = this.items.find(x => x.type === "feat" && x.data.data.requirements?.toLowerCase() === "condition" && x.name.toLowerCase() === conditionName.toLowerCase());
+        const conditionItem = this.items.find(item => this._isCondition(item) && item.name.toLowerCase() === conditionName.toLowerCase());
         return conditionItem;
     }
 
-    async setCondition(conditionName, enabled) {
-        if (!SFRPG.statusEffectIconMapping[conditionName]) {
-            ui.notifications.warn(`Trying to set condition ${conditionName} on actor ${this.name} but the condition is not valid. See CONFIG.SFRPG.statusEffectIconMapping for all valid conditions.`);
+    /**
+     * Get an Array of all conditions on the actor.
+     * @returns {Array} strings
+     * @public
+     */
+    getActiveConditions() {
+        const activeConditions = this.items.filter(item => this._isCondition(item));
+        return activeConditions;
+    }
+
+    /**
+     * Checks if the item is a feat containing the data requirements of condition.
+     * @param item foundry item document
+     * @returns {boolean}
+     * @private
+     */
+    _isCondition(item) {
+        return item.type === "feat" && item.data.data.requirements?.toLowerCase() === "condition";
+    }
+
+    _isStatusEffect(name) {
+        return SFRPG.statusEffects.find(effect => effect.id === name) != undefined;
+    }
+
+    /**
+     * Updates the Actor's conditions. Either adds or removes a condition Item as necessary to match the enabled argument.
+     * @param {String} conditionName The name of the condition. Must match any key from config.js SFRPG.statusEffects. Case sensitive.
+     * @param {Boolean} enabled If this value is true it ensures the condition is present on the Actor.
+     * @param {Object} overlay If this value is true it indicates that the token icon should be added as a full-sized overlay. Default is false.
+     * @returns {Promise<*>} The Promise resulting from the create or delete Embedded Document call.
+     * @public
+     */
+    async setCondition(conditionName, enabled, { overlay = false } = {}) {
+        if (!this._isStatusEffect(conditionName)) {
+            ui.notifications.warn(`Trying to set condition ${conditionName} on actor ${this.name} but the condition is not valid. See CONFIG.SFRPG.statusEffects for all valid conditions.`);
             return;
         }
 
-        const conditionItem = this.getCondition(conditionName);
+        // Try to get status effect object as a workaround for a poorly conceived check in foundry.js Token.toggleEffect(...)
+        let statusEffect = SFRPG.statusEffects.find(effect => effect.id === conditionName);
```

Comment:
const instead of let please. I don't think statusEffect is altered anywhere inside the function.